### ISSUE TITLE
Format SCSS deep map like CSS rules 

### ISSUE THREAD

**Issue by ismail-syed** [prettier/prettier#3235]
**Prettier 1.8.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASAtgQwA5IAQAUkANhAE74DEAjLXQDR7wAeMAtAM4AWmAJhAHd81bMzwixABjwdMxdNACUAbgA6UdahhcycPhzbYyEbHDIwAlnA74s2NujMBzOAXV48WnXt4GjJs0trRjcoDw9UACNyXjM2YkxIuGJ8AkV3PHSoFRB6EBNLaA5kUCMLWAB1C15tZAAOSTyYRKqarmQAJjyOcqdiOABFAFcIeGQAMzkOOCayTAtiXoBhCHQsZBAoaDhckEi5sABrOBgAZWxMMF7kGDIhmZAAKw5mACED47PMRwAZcp2kJNiNM8hcyNMyBswBxit04OgLDc7g8htMACqJYqAqYPcoQmAABTmTnW2OBD10AEchhZdETMCTMBMcaDjNMKnNsMzyXl+GBuSCQJgODABQ8IEMYNhJR0xQBfOVAA)
```sh
--parser css
--no-prose-wrap
```

**Input:**
```css
$map: (color: #111111, text-shadow: 1px 1px 0 salmon);

$threads-properties: map-merge(
  $threads-properties, (
    $border-label: ()
  )
);
```

**Output:**
```css
$map: (
  color: #111111,
  text-shadow: 1px 1px 0 salmon
);

$threads-properties: map-merge($threads-properties,  ($border-label: ()));

```

**Expected behavior:**
`$threads-properties:` code block to be formatted as is, over multiple lines

**Actual behavior:**
It formats the `$threads-properties:` to one line.

**Comment by ismail-syed** [comment]
cc/ @asmockler @beefchimi

**Comment by e-cloud** [comment]
another not good enough format case:

input:
```scss
$dt-teal: (
  50: #e1f2f1,
  100: #b8e6e2,
  200: #82d9d2,
  300: #5cd1c7,
  400: #46c7bc,
  500: #36b6ac,
  600: #2e9990,
  700: #268078,
  800: #1f6660,
  900: #174d48,
  A100: #b2ffec,
  A200: #66ffd9,
  A400: #23ebc3,
  A700: #00bfac,
  contrast: (
    50: $black-87-opacity,
    100: $black-87-opacity,
    200: $black-87-opacity,
    300: $black-87-opacity,
    400: $black-87-opacity,
    500: white,
    600: white,
    700: white,
    800: $white-87-opacity,
    900: $white-87-opacity,
    A100: $black-87-opacity,
    A200: $black-87-opacity,
    A400: $black-87-opacity,
    A700: $black-87-opacity,
  )
);
```

output:
```scss
$dt-teal: (
  50: #e1f2f1,
  100: #b8e6e2,
  200: #82d9d2,
  300: #5cd1c7,
  400: #46c7bc,
  500: #36b6ac,
  600: #2e9990,
  700: #268078,
  800: #1f6660,
  900: #174d48,
  A100: #b2ffec,
  A200: #66ffd9,
  A400: #23ebc3,
  A700: #00bfac,
  contrast:
    (
      50: $black-87-opacity,
      100: $black-87-opacity,
      200: $black-87-opacity,
      300: $black-87-opacity,
      400: $black-87-opacity,
      500: white,
      600: white,
      700: white,
      800: $white-87-opacity,
      900: $white-87-opacity,
      A100: $black-87-opacity,
      A200: $black-87-opacity,
      A400: $black-87-opacity,
      A700: $black-87-opacity
    )
);
```

an expected result should be the same as input.

**Comment by e-cloud** [comment]
And in @ismail-syed output, there is another problem:

```scss
$threads-properties: map-merge($threads-properties,  ($border-label: ()));
                                                   ^^
```
it has redundant white spaces.

**Comment by mrliptontea** [comment]
Also, nested maps ideally should be formatted consistently with the same wrapping, although I imagine it could be hard.

Example (with Prettier 1.9.2) - ideally output formatting should remain the same.
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASCAjAVnMMC0AzgJYBechSABABQA6UVTVMxUAntfY87wA4BDACZC2Ac3wAPagAYAdACYATnAC2AGga9+w0VAmcq8gIwBWFRq3amAM2gES5aqjuwiZOPlYcrTAJSaPEyEqgIANmFcvtqCIuJSsooAbADsFoHWTLF6BokKqenRvK4OHs4l7uREoRHRAdFhAkpicFFBMbrx0lTGhe06cfr4hvJJBWoZmRWOrVQu9pWejc1wdQx+ANwMIOogEHys0ITIoE1KEADuAApNCMco4RcC7Me7AoQwyDBKAK5wuxglAIwABrOAwADKgjA4i+v3+ICEEDAyBs4UICLYGKUMCuQLEoVR6IRWEIkgAQkDQeCIQJVHAADJsOBEsIY3YQH4wPhchSs9kgQRKbHIECEMCEV6CpRsGAAdWIQhgAAtkAAOGS7PjnDFyoF8UXaihwJQANxZuxUAEcfsQVHiBASBPyERjVMQ4X9diR9GE4ABFH4QeAu3YwAQYBVK1VIAAsYaBxDC4gAwhBVISHhEdiAfhiACoR+5otlwAC+ZaAA)

**Input:**

```scss
$object-sizes: (
    tiny: (
        padding-x: 0.2rem,
        padding-y: 0.15rem,
        font-size: $font-size-tiny
    ),
    small: (
        padding-x: 0.267rem,
        padding-y: 0.267rem,
        font-size: $font-size-small
    ),
    large: (
        padding-x: 1rem,
        padding-y: 0.667rem,
        font-size: $font-size-large
    )
);
```

**Output:**

```scss
$object-sizes: (
    tiny: (padding-x: 0.2rem, padding-y: 0.15rem, font-size: $font-size-tiny),
    small:
        (
            padding-x: 0.267rem,
            padding-y: 0.267rem,
            font-size: $font-size-small
        ),
    large: (padding-x: 1rem, padding-y: 0.667rem, font-size: $font-size-large)
);
```

**Comment by evilebottnawi** [comment]
Try to fix it for `1.13` :+1:
